extensions/vscode/src/extension.ts

```diff
@@ -151,6 +148,32 @@ export async function activate(context: ExtensionContext) {
   context.subscriptions.push(new PublisherAuthProvider(state));
 }
 
+// This method is called when your extension is activated
+// Your extension is activated the very first time the command is executed
+export function activate(context: ExtensionContext) {
+  console.log("Posit Publisher extension activated");
+  // Is our workspace trusted?
+  if (workspace.isTrusted) {
+    console.log("initializing extension within a trusted workspace");
+    initializeExtension(context);
+    return;
+  }
+  console.log(
+    "activated within a restricted workspace. Initialization is paused until trust is granted.",
+  );
+
+  // We are not trusted yet... so if and when we are, then start the initialization
+  context.subscriptions.push(
+    workspace.onDidGrantWorkspaceTrust(() => {
+      console.log("Trust mode granted by user.");
+      console.log(
+        "Proceeding forward with initializion for extension within a trusted workspace",
+      );
+      initializeExtension(context);
+    }),
+  );
```

Comment:
Do we need to dispose of this listener when the workspace is granted trust?

Reply:
No, it appears to only be happening once. As you saw below, VSCode reloads the extensions when trust is taken away.